### PR TITLE
Task/add fields to db allow reupload

### PR DESCRIPTION
**Pull request by itraviv**

### CODE REVIEW

File: lms/lmsdb/bootstrap.py [hunk 1]
```diff
@@ -1,37 +1,64 @@
-from peewee import ProgrammingError
+from typing import Type
 
+from peewee import Model, Field
 from playhouse.migrate import migrate
 
-from lms.lmsdb import database_config  # noqa: I100
+from lms.lmsdb import database_config, database  # noqa: I100
```
Comment (yammesicka) [inline lms/lmsdb/bootstrap.py:6]: database should appear before database_config

File: lms/lmsdb/bootstrap.py [hunk 2]
```diff
@@ -1,37 +1,64 @@
...
-        with database_config.database.atomic():
-            models.CommentText.create_comment('dummy222', 'dummy222')
-            database_config.database.rollback()
-    except ProgrammingError:
-        exists = False
-        database_config.database.close()
-
-    if exists:
-        print('No need to create flake8_key')  # noqa: T001
+def _migrate_column_in_table_if_needed(
```
Comment (yammesicka) [inline lms/lmsdb/bootstrap.py:11]: Excellent

File: lms/lmsweb/views.py [hunk 3]
```diff
@@ -180,11 +180,18 @@ def fetch_solutions(user_id):
...
-        .join(Solution, 'LEFT OUTER', on=solutions_filter)
-        .where(Exercise.is_archived == False)  # NOQA: E712
-        .order_by(Exercise.id)
+            .select(*fields)
+            .join(Solution, 'LEFT OUTER', on=solutions_filter)
+            .where(Exercise.is_archived == False)  # NOQA: E712
+            .group_by(Exercise)
+            .order_by(Exercise.id)
     )
+    if Solution.select().count():
```
Comment (yammesicka) [inline lms/lmsweb/views.py:189]: This is extremely expensive to put on our main page. Why is it actually needed?

**Comment by yammesicka** [comment]
We should really ship this feature already :(
Please merge and deploy ASAP, fix the problems in my CR afterwards.

File: lms/lmsdb/bootstrap.py [hunk 4]
```diff
@@ -1,37 +1,64 @@
...
+    table: Type[Model],
+    field_instance: Field,
+):
+    column_name = field_instance.name
+    table_name = table.__name__.lower()
+    cols = {col.name for col in database.get_columns(table_name)}
```
Comment (gal432) [inline lms/lmsdb/bootstrap.py:17]: use database_config.database and remove the import database

File: lms/lmsweb/views.py [hunk 5]
```diff
@@ -328,32 +335,23 @@ def upload():
...
         desc = "did you use Upload <number of exercise> ? (example: Upload 1)"
         return fail(402, f'{msg}, {desc}')
-    matches, misses, duplications = set(), set(), set()
+    matches, misses = set(), set()
     for exercise_id, code in exercises:
         exercise = Exercise.get_or_none(Exercise.id == exercise_id)
         if exercise is None:
             misses.add(exercise_id)
             continue
-        solution, created = Solution.create_solution(
```
Comment (gal432) [inline lms/lmsweb/views.py:337]: why not using create_solution ?

**Pull request merged**